frontend/src/metabase/public/containers/PublicDashboard/PublicDashboard.tsx

```diff
@@ -90,16 +79,28 @@ const connector = connect(mapStateToProps, mapDispatchToProps);
 type ReduxProps = ConnectedProps<typeof connector>;
 
 type OwnProps = {
-  params: {
-    dashboardId?: DashboardId;
-    uuid?: string;
-    token?: string;
-  };
-} & DashboardControlsProps;
+  dashboardId: DashboardId;
+  queryParams: Record<string, string | string[] | null | undefined>;
+};
+
+type DisplayProps = Pick<
+  DashboardDisplayOptionControls,
+  | "isFullscreen"
+  | "isNightMode"
+  | "onFullscreenChange"
+  | "onNightModeChange"
+  | "onRefreshPeriodChange"
+  | "refreshPeriod"
+  | "setRefreshElapsedHook"
+  | "hasNightModeToggle"
+>;
 
-type PublicDashboardProps = ReduxProps &
-  OwnProps &
-  DashboardControlsPassedProps;
+type PublicDashboardProps = OwnProps &
+  ReduxProps &
+  DisplayProps &
+  EmbedDisplayParams & {
+    queryParams: Record<string, string | string[] | null | undefined>;
```

Comment:
(I wonder if we should give `Record<string, string | string[] | null | undefined>` a name. I like that it's explicit here, but we have a lot of usages that it might be worth it to create a new type.)

Reply:
Yeah that's fair, let me see if there's something I can use here

Reply:
Ah! we can use `Query` from `history`. It's the expected type anyway when we're dealing with location query parameters. I think I'd like to narrow this for the StaticDashboard at some point but for now this is the type we're passing from other components to `PublicDashboard` so I'll leave it